src/control/positionControl/pathPlanning/NumTreesPlanning.cpp

```diff
@@ -0,0 +1,214 @@
+//
+// Created by ratoone on 09-12-19.
+//
+
+#include "control/positionControl/pathPlanning/NumTreesPlanning.h"
+
+namespace rtt::ai::control{
+
+NumTreesPlanning::NumTreesPlanning(const CollisionDetector &collisionDetector) :
+    collisionDetector(collisionDetector) {}
+
+std::list<rtt::Vector2>
+NumTreesPlanning::computePath(const rtt::Vector2 &robotPosition, const rtt::Vector2 &targetPosition) {
+    DT = 0.3 / rtt::ai::GameStateManager::getCurrentGameState().getRuleSet().maxRobotVel;
+    if (DT > 0.12) DT = 0.12;
+    if (DT < 0.06) DT = 0.06;
+
+    auto path = tracePath(robotPosition, targetPosition);
+    double goToTimeInFuture = 0.4;
+    auto targetPathPoint = static_cast<unsigned long>(goToTimeInFuture/DT);
+    if (path.size() < targetPathPoint) {
+        return std::list<Vector2>(1, targetPosition);
+    }
+    return path;
+}
+
+std::list<Vector2> NumTreesPlanning::tracePath(const rtt::Vector2 &currentPosition, const rtt::Vector2 &targetPosition) {
```

Comment:
This function is a 100 lines of code. If you're going to refactor it please reduce it to ~30 max.

Reply:
Although feel free to disregard this as you did not change much it seems.

Reply:
I didn't change anything from the initial mess. I could do some cleanup though

Reply:
More cleanup scheduled with #777